templates/etc/keepalived/keepalived.conf.j2

```diff
@@ -141,9 +141,19 @@ vrrp_instance {{ key }} {
   }
 {% endif %}
 
+{% if value.vmac_xmit_base is defined and value.vmac_xmit_base | bool %}
+  vmac_xmit_base
+{% endif %}
+
+{% if value.unicast_src_ip is defined %}
+  unicast_src_ip {{ value.unicast_src_ip }}
+{% endif %}
+
 {% if value.unicast_peer is defined %}
   unicast_peer {
-    {{ value.unicast_peer }}
+{% for peer in value.unicast_peer %}
```

Comment:
This is not backwards compatible

Reply:
Maybe do a type check? And handle both type correcty.

Reply:
Do you have an example what was there with multiple entries before? With single entry it should be identical. I couldn't get it running, so this is why i modifed it.

Reply:
I'll habe a look

Reply:
Something like this:

```
{% if frontend.use_backend is iterable and frontend.use_backend is not string %}
{% for use_backend in frontend.use_backend | default([]) %}
  use_backend {{ use_backend }}
{% endfor %}
{% else %}
  use_backend {{ frontend.use_backend }}
{% endif %}
```

or
 
```
{% if postfix_inet_interfaces is string %}
inet_interfaces = {{ postfix_inet_interfaces }}
{% else %}
inet_interfaces = {{ postfix_inet_interfaces | join(', ') }}
{% endif %}
```

might be enough

Reply:
can you take a look on the new commit if it suits better?

Reply:
Yes

Reply:
It should look like this:

```diff
# git diff
diff --git a/templates/etc/keepalived/keepalived.conf.j2 b/templates/etc/keepalived/keepalived.conf.j2
index 90f150c..a27fceb 100644
--- a/templates/etc/keepalived/keepalived.conf.j2
+++ b/templates/etc/keepalived/keepalived.conf.j2
@@ -143,7 +143,13 @@ vrrp_instance {{ key }} {
 
 {% if value.unicast_peer is defined %}
   unicast_peer {
+{% if value.unicast_peer is string %}
     {{ value.unicast_peer }}
+{% else %}
+{% for peer in value.unicast_peer %}
+    {{ peer }}
+{% endfor%}
+{% endif %}
   }
 {% endif %}
```

Reply:
Done